apps/studio/src/server/modules/resource/resource.service.ts

```diff
@@ -409,3 +410,183 @@ export const publishResource = async (
 
   return addedVersionResult
 }
+
+export const getWithFullPermalink = async ({
```

Comment:
this is refactored and extracted out from an existing code in resource.router so that its reusable